## Exploratory Data Analysis of Superstore 

Exploratory data analysis has been performed on the Superstore [dataset](https://drive.google.com/file/d/1lV7is1B566UQPYzzY8R2ZmOritTW299S/view) using pandas.

### Importing pandas libraray

In [224]:
import pandas as pd

### Importing the dataset

In [225]:
df = pd.read_csv('SampleSuperstore.csv')
print('Data has been imported!')
df.head()

Data has been imported!


Ship Mode    Segment        Country             City       State  \
0    Second Class   Consumer  United States        Henderson    Kentucky   
1    Second Class   Consumer  United States        Henderson    Kentucky   
2    Second Class  Corporate  United States      Los Angeles  California   
3  Standard Class   Consumer  United States  Fort Lauderdale     Florida   
4  Standard Class   Consumer  United States  Fort Lauderdale     Florida   

   Postal Code Region         Category Sub-Category     Sales  Quantity  \
0        42420  South        Furniture    Bookcases  261.9600         2   
1        42420  South        Furniture       Chairs  731.9400         3   
2        90036   West  Office Supplies       Labels   14.6200         2   
3        33311  South        Furniture       Tables  957.5775         5   
4        33311  South  Office Supplies      Storage   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164

### Checking the basic info 

In [226]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Ship Mode     9994 non-null   object 
 1   Segment       9994 non-null   object 
 2   Country       9994 non-null   object 
 3   City          9994 non-null   object 
 4   State         9994 non-null   object 
 5   Postal Code   9994 non-null   int64  
 6   Region        9994 non-null   object 
 7   Category      9994 non-null   object 
 8   Sub-Category  9994 non-null   object 
 9   Sales         9994 non-null   float64
 10  Quantity      9994 non-null   int64  
 11  Discount      9994 non-null   float64
 12  Profit        9994 non-null   float64
dtypes: float64(3), int64(2), object(8)
memory usage: 1015.1+ KB


The dataset have **9994 entries** and **12 columns**.

### Descriptive statistics

In [227]:
df.describe()

Postal Code         Sales     Quantity     Discount       Profit
count   9994.000000   9994.000000  9994.000000  9994.000000  9994.000000
mean   55190.379428    229.858001     3.789574     0.156203    28.656896
std    32063.693350    623.245101     2.225110     0.206452   234.260108
min     1040.000000      0.444000     1.000000     0.000000 -6599.978000
25%    23223.000000     17.280000     2.000000     0.000000     1.728750
50%    56430.500000     54.490000     3.000000     0.200000     8.666500
75%    90008.000000    209.940000     5.000000     0.200000    29.364000
max    99301.000000  22638.480000    14.000000     0.800000  8399.976000

### Checking for null/missing values

In [228]:
df.isnull().sum()

Ship Mode       0
Segment         0
Country         0
City            0
State           0
Postal Code     0
Region          0
Category        0
Sub-Category    0
Sales           0
Quantity        0
Discount        0
Profit          0
dtype: int64

There is **no null/missing value** in the dataset.

### Data Analysis

In [229]:
#checking for different Countries
df['Country'].value_counts()

United States    9994
Name: Country, dtype: int64

Since there is only one Country, dropping the Country column.

In [230]:
df.drop(['Country'],axis=1,inplace=True)

In [231]:
#new dataset after dropping Country column
df.head(5)

Ship Mode    Segment             City       State  Postal Code Region  \
0    Second Class   Consumer        Henderson    Kentucky        42420  South   
1    Second Class   Consumer        Henderson    Kentucky        42420  South   
2    Second Class  Corporate      Los Angeles  California        90036   West   
3  Standard Class   Consumer  Fort Lauderdale     Florida        33311  South   
4  Standard Class   Consumer  Fort Lauderdale     Florida        33311  South   

          Category Sub-Category     Sales  Quantity  Discount    Profit  
0        Furniture    Bookcases  261.9600         2      0.00   41.9136  
1        Furniture       Chairs  731.9400         3      0.00  219.5820  
2  Office Supplies       Labels   14.6200         2      0.00    6.8714  
3        Furniture       Tables  957.5775         5      0.45 -383.0310  
4  Office Supplies      Storage   22.3680         2      0.20    2.5164

In [232]:
#checking number of unique values in each column
for col in list(df):
    print(col, df[col].nunique())

Ship Mode 4
Segment 3
City 531
State 49
Postal Code 631
Region 4
Category 3
Sub-Category 17
Sales 5825
Quantity 14
Discount 12
Profit 7287


In [233]:
#different ship mode
df['Ship Mode'].value_counts()

Standard Class    5968
Second Class      1945
First Class       1538
Same Day           543
Name: Ship Mode, dtype: int64

In [234]:
#different regions
df['Region'].value_counts()

West       3203
East       2848
Central    2323
South      1620
Name: Region, dtype: int64

In [235]:
#different segments
df['Segment'].value_counts()

Consumer       5191
Corporate      3020
Home Office    1783
Name: Segment, dtype: int64

In [236]:
#different categories
df['Category'].value_counts()

Office Supplies    6026
Furniture          2121
Technology         1847
Name: Category, dtype: int64

In [237]:
#different sub-categories
df['Sub-Category'].value_counts()

Binders        1523
Paper          1370
Furnishings     957
Phones          889
Storage         846
Art             796
Accessories     775
Chairs          617
Appliances      466
Labels          364
Tables          319
Envelopes       254
Bookcases       228
Fasteners       217
Supplies        190
Machines        115
Copiers          68
Name: Sub-Category, dtype: int64

In [238]:
#total sales and profit
print('Total Sales: {}'.format(df['Sales'].sum()))
print('Total Profit: ', df['Profit'].sum())

Total Sales: 2297200.8603000003
Total Profit:  286397.0217


In [239]:
#sum sales and profit from each category
df.groupby('Category')[['Sales','Profit']].sum()

Sales       Profit
Category                                 
Furniture        741999.7953   18451.2728
Office Supplies  719047.0320  122490.8008
Technology       836154.0330  145454.9481

In [240]:
#sum sales and profit from each sub-category
df.groupby('Sub-Category')[['Sales','Profit']].sum()

Sales      Profit
Sub-Category                         
Accessories   167380.3180  41936.6357
Appliances    107532.1610  18138.0054
Art            27118.7920   6527.7870
Binders       203412.7330  30221.7633
Bookcases     114879.9963  -3472.5560
Chairs        328449.1030  26590.1663
Copiers       149528.0300  55617.8249
Envelopes      16476.4020   6964.1767
Fasteners       3024.2800    949.5182
Furnishings    91705.1640  13059.1436
Labels         12486.3120   5546.2540
Machines      189238.6310   3384.7569
Paper          78479.2060  34053.5693
Phones        330007.0540  44515.7306
Storage       223843.6080  21278.8264
Supplies       46673.5380  -1189.0995
Tables        206965.5320 -17725.4811

In [241]:
#max discount given in each category
df.groupby('Category').max()['Discount'].nlargest()

Category
Office Supplies    0.8
Furniture          0.7
Technology         0.7
Name: Discount, dtype: float64

In [242]:
#max discount given in each sub-category
df.groupby('Sub-Category').max()['Discount'].nlargest()

Sub-Category
Appliances     0.8
Binders        0.8
Bookcases      0.7
Machines       0.7
Furnishings    0.6
Name: Discount, dtype: float64

In [243]:
#top 5 cities with maximum sales
df.groupby('City').max()['Sales'].nlargest(5)

City
Jacksonville     22638.480
Lafayette        17499.950
Seattle          13999.960
New York City    11199.968
Newark           10499.970
Name: Sales, dtype: float64

In [244]:
#lowest 5 cities with maximum sales
df.groupby('City').max()['Sales'].nsmallest(5)

City
Abilene         1.392
Elyria          1.824
Jupiter         2.064
Pensacola       2.214
Ormond Beach    2.808
Name: Sales, dtype: float64

In [245]:
#max profit from each segment
df.groupby('Segment').max()['Profit'].nlargest()

Segment
Corporate      8399.9760
Consumer       6719.9808
Home Office    3919.9888
Name: Profit, dtype: float64

Most profitable segment is **Corporate**.

In [258]:
#sales and profit by cities
df.groupby('City')[['Sales','Profit']].sum().nlargest(10,'Sales')

Sales      Profit
City                                  
New York City  256368.1610  62036.9837
Los Angeles    175851.3410  30440.7579
Seattle        119540.7420  29156.0967
San Francisco  112669.0920  17507.3854
Philadelphia   109077.0130 -13837.7674
Houston         64504.7604 -10153.5485
Chicago         48539.5410  -6654.5688
San Diego       47521.0290   6377.1960
Jacksonville    44713.1830  -2323.8350
Springfield     43054.3420   6200.6974

The largest amount of sales and profit has been obtained from **New York City**, followe by **Los Angeles**, **Seattle** and **San Francisco**.

In [249]:
#top 5 profitable states
df.groupby('State')[['Sales','Profit']].sum().nlargest(5,'Profit')

Sales      Profit
State                              
California  457687.6315  76381.3871
New York    310876.2710  74038.5486
Washington  138641.2700  33402.6517
Michigan     76269.6140  24463.1876
Virginia     70636.7200  18597.9504

In [250]:
#least 5 profitable states
df.groupby('State')[['Sales','Profit']].sum().nsmallest(5,'Profit')

Sales      Profit
State                                  
Texas           170188.0458 -25729.3563
Ohio             78258.1360 -16971.3766
Pennsylvania    116511.9140 -15559.9603
Illinois         80166.1010 -12607.8870
North Carolina   55603.1640  -7490.9122

### The analysis has been done!